ethereum/permissioning/src/main/java/tech/pegasys/pantheon/ethereum/permissioning/node/NodePermissioningController.java

```diff
@@ -25,22 +25,28 @@
   private static final Logger LOG = LogManager.getLogger();
 
   private final Optional<SyncStatusNodePermissioningProvider> syncStatusNodePermissioningProvider;
+  NodePermissioningProvider[] providers;
 
   public NodePermissioningController(
```

Comment:
The `NodePermissiningController` should also be able to read the permissioning config flags and use the `NodePermissioningProviders` accordingly.

It feels like we could use a factory that from the PermissioningConfiguration gives you an instance of `NodePermissioningController`.

Reply:
there is a certain dependency on the CLI options for smartContractAddress and nodeEnabled/accountEnabled

Reply:
I've added a factory - but not sure how to test it without exposing the providers[] directly

Reply:
also added a separate SmartContractPermissioningConfiguration object - made more sense than adding to PermissioningConfiguration which encapsulates the TOML config (and gets reloaded from file) and I think PermissioningConfiguration should be refactored to LocalPermissioningConfiguration but wanted to validate this approach (two separate configs) first

Reply:
I think we should have one PermissioningConfiguration object. From what I remember, the PermissioningConfigurationBuilder is the one that does all the TOML magic.

Reply:
ok I made one PermissioningConfiguration class that has two optionals - LocalPermissioningConfiguration and SmartContractPermissioningConfiguratoin

Reply:
sorry about all the refactoring...